app/components/orangelight/advanced_search_form_component.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class Orangelight::AdvancedSearchFormComponent < Blacklight::AdvancedSearchFormComponent
+  def sort_fields_select
+    options = sort_fields.values.map { |field_config| [helpers.sort_field_label(field_config.key), field_config.key] }
+    return unless options.any?
+
+    select_tag(:sort, options_for_select(options, params[:sort]), class: "sort-select", aria: { labelledby: 'advanced-search-sort-label' })
+  end
+
+  def initialize_search_filter_controls
+    fields = blacklight_config.facet_fields.select { |_k, v| v.include_in_advanced_search || v.include_in_advanced_search.nil? }
+
+    fields.each do |_k, config|
+      config.advanced_search_component = Orangelight::FacetFieldCheckboxesComponent
+      display_facet = @response.aggregations[config.field]
+      with_search_filter_control(config:, display_facet:)
+    end
+  end
+
+  def pub_date_field
+    blacklight_config.facet_fields['pub_date_start_sort']
+  end
+
+  def pub_date_presenter
+    view_context.facet_field_presenter(pub_date_field, {})
+  end
+
+  def initialize_search_field_controls
+    search_fields.values.each.with_index do |field, index|
+      with_search_field_control do
+        fields_for_etc(index:, field:)
```

Comment:
Could the method get a more descriptive name, e.g.

```suggestion
        generate_fields(index:, field:)
```

Reply:
Committed